diff --git a/src/detection/opengl/opengl_linux.c b/src/detection/opengl/opengl_linux.c
index 5f88b7e..0f76387 100644
--- a/src/detection/opengl/opengl_linux.c
+++ b/src/detection/opengl/opengl_linux.c
@@ -4,7 +4,7 @@
 
 #include <string.h>
 
-#if __ANDROID__ && !defined(FF_HAVE_EGL)
+#if __ANDROID__ && !defined(FF_HAVE_EGL) && !FF_MODULE_DISABLE_OPENGL
     // On Android, installing OpenGL headers is enough (mesa-dev)
     #if __has_include(<EGL/egl.h>)
         #define FF_HAVE_EGL 1
